model/src/main/java/org/projectnessie/model/Merge.java

```diff
@@ -40,6 +43,10 @@
   @JsonFormat(pattern = Validation.HASH_REGEX)
   String getFromHash();
 
+  @NotBlank
```

Comment:
what are the rules for the null-ness of from hash and from refname? Would it make sense to have this as `Reference fromRef`?

Reply:
Not here probably. But `org.projectnessie.client.api.MergeReferenceBuilder` has both variants:
* either specify just `fromRefName(String)` to assume HEAD
* or `fromRef(Reference)` (which is technically a convenience for `fromRefName(fromRef.getName()).fromHash(fromRef.getHash())`).

Reply:
I.e. you get that via `NessieApiV!`